Aquarius/Src/Aquarius/Renderer/Shader.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <iostream>
+#include <glm/glm.hpp>
+#include <unordered_map>
+
+namespace Aquarius {
+
+    class Shader
+    {
+    public:
+        // Path to vertex and fragment shader source code
+        Shader(const std::string& vertexPath, const std::string& fragmentPath);
+        ~Shader();
+
+        // Enable/disable this shader
+        void activate();
+        void deactivate();
+
+        // Set uniform methods
+        void setBool(const std::string& name, bool value) const;
+        void setInt(const std::string& name, int value) const;
+        void setInt2(const std::string& name, const glm::ivec2& values) const;
+        void setInt3(const std::string& name, const glm::ivec3& values) const;
+        void setInt4(const std::string& name, const glm::ivec4& values) const;
+        void setFloat(const std::string& name, float value) const;
+        void setFloat2(const std::string& name, const glm::vec2& values) const;
+        void setFloat3(const std::string& name, const glm::vec3& values) const;
+        void setFloat4(const std::string& name, const glm::vec4& values) const;
+        void setMat4(const std::string& name, bool transpose, const glm::mat4& value) const;
+
+        enum class ShaderType
+        {
+            VERTEX_SHADER,
+            FRAGMENT_SHADER
+        };
+
+        std::unordered_map<ShaderType, std::string> shaderTypeToString =
+        {
+            {ShaderType::VERTEX_SHADER, "VERTEX_SHADER"},
+            {ShaderType::FRAGMENT_SHADER, "FRAGMENT_SHADER"}
+        };
```

Comment:
Could probably be `static const`

Reply:
Yes very much so, I was thinking static for sure but const didnt even cross my mind

Reply:
When I implemented this, I learned a lot! Init'ing the static member outside the header, then not being able to use the [] operator because it isn't const, I had to use .at()!